processor/k8sprocessor/factory.go

```diff
@@ -191,3 +194,26 @@ func createProcessorOpts(cfg config.Processor) []Option {
 
 	return opts
 }
+
+func warnDeprecatedMetadataConfig(logger *zap.Logger, cfg config.Processor) {
+	oCfg := cfg.(*Config)
+
+	for _, field := range oCfg.Extract.Metadata {
+		switch field {
+		case metdataNamespace:
+			logger.Warn("metdataNamespace is a deprecated configuration for k8s-tagger processor")
```

Comment:
1. We should use the identifier users would use in config in the log message, not the variable name. `metadataNamespace` should be `namespace` (value of metadataNamespace var).
2. The message should also mention the new value that replaces the deprecated one.

The log message should look like:

> metadata: `namespace` has been deprecated in favor of `k8s.namespace.name`.

This will allow users to quickly identify the config option they need to change and the value they should change it to.